presto-main/src/main/java/com/facebook/presto/operator/aggregation/reservoirsample/UnweightedDoubleReservoirSample.java

```diff
@@ -0,0 +1,162 @@
+    {
+        this.count = count;
+        this.samples = samples;
+    }
+
+    public int getMaxSamples()
+    {
+        return samples.length;
+    }
+
+    public void add(double element)
+    {
+        if (++count <= samples.length) {
+            samples[count - 1] = element;
+            return;
+        }
+        int index = ThreadLocalRandom.current().nextInt(0, count);
+        if (index < samples.length) {
+            samples[index] = element;
+        }
+    }
+
+    public void mergeWith(UnweightedDoubleReservoirSample other)
+    {
+        checkArgument(samples.length == other.samples.length, "Max samples must be equal");
+        if (other.count < other.samples.length) {
+            other.stream().forEach(o -> add(o));
+            return;
+        }
+        if (count < samples.length) {
+            final UnweightedDoubleReservoirSample target = ((UnweightedDoubleReservoirSample) other.clone());
+            stream().forEach(o -> target.add(o));
+            count = target.count;
+            samples = target.samples;
+            return;
+        }
+        shuffleArray(samples);
+        shuffleArray(other.samples);
+        int nextIndex = 0;
+        int otherNextIndex = 0;
+        double[] merged = new double[samples.length];
+        for (int i = 0; i < samples.length; ++i) {
+            if (ThreadLocalRandom.current().nextLong(0, count + other.count) < count) {
+                merged[i] = samples[nextIndex++];
+            }
+            else {
+                merged[i] = other.samples[otherNextIndex++];
+            }
+        }
+        count += other.count;
+        samples = merged;
+    }
+
+    @Override
+    public UnweightedDoubleReservoirSample clone()
+    {
+        return new UnweightedDoubleReservoirSample(this);
+    }
+
+    public DoubleStream stream()
+    {
+        int effectiveSize = Math.min(count, samples.length);
+        return Arrays.stream(samples, 0, effectiveSize);
+    }
+
+    private static void shuffleArray(double[] ar)
+    {
+        for (int i = ar.length - 1; i > 0; i--) {
+            int index = ThreadLocalRandom.current().nextInt(0, i + 1);
+            double a = ar[index];
+            ar[index] = ar[i];
+            ar[i] = a;
+        }
+    }
+
+    public static UnweightedDoubleReservoirSample deserialize(SliceInput input)
+    {
+        int count = input.readInt();
+        int maxSamples = input.readInt();
+        double[] samples = new double[maxSamples];
+        input.readBytes(
+                Slices.wrappedDoubleArray(samples),
+                Math.min(count, samples.length) * SizeOf.SIZE_OF_DOUBLE);
+        return new UnweightedDoubleReservoirSample(count, samples);
+    }
+
+    public void serialize(SliceOutput out)
+    {
+        out.appendInt(count);
+        out.appendInt(samples.length);
+        IntStream.range(0, Math.min(count, samples.length)).forEach(i -> out.appendDouble(samples[i]));
```

Comment:
use a for loop here

Reply:
Searched through the code and removed all`IntStream` instances.